### PR TITLE
optimize: configure registry access credential via sdk

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

1. configure registry access credential via sdk, gen config.json locally and send to all hosts which needed to access registry.

### CODE REVIEW

**Review by VinceCui — COMMENTED** [review]
Where initial the c.Auth?

**Review by VinceCui — COMMENTED** [review]
please resolve the conflict

File: pkg/registry/singleton.go [hunk 1]
```diff
@@ -281,24 +280,60 @@ func (c *localSingletonConfigurator) configureHostsFile(hosts []net.IP) error {
...
+			logrus.Debugf("failed to remove tmp registry auth file:%s", tmpAuthFilePath)
+		}
+	}()
+
+	err = c.copy2RemoteHosts(tmpAuthFilePath, c.containerRuntimeInfo.ConfigFilePath, hosts)
+	if err != nil {
+		return err
+	}
+
+	err = c.copy2RemoteHosts(tmpAuthFilePath, remoteKubeletAuthFilePath, hosts)
```
Comment (VinceCui) [inline pkg/registry/singleton.go:324]: mkdir /var/lib/kubelet firstly, because registry is before kube-runtime

Comment (kakaZhou719) [reply pkg/registry/singleton.go:324]: copy2RemoteHosts will mkdir,  if remote dir is not exist .

**Comment by kakaZhou719** [comment]
> Where initial the c.Auth?

cluster runtime installer.go will init registry config include auth info.

**Comment by VinceCui** [comment]
> > Where initial the c.Auth?
> 
> cluster runtime installer.go will init registry config include auth info.

@kakaZhou719 which PR did this?

**Review by VinceCui — APPROVED** [review]
LGTM

**Pull request merged**